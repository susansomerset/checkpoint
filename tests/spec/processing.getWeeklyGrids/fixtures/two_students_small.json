{
  "studentData": {
    "students": [
      {
        "id": "S1",
        "name": "Alice",
        "courses": [
          {
            "id": "C-101",
            "name": "Algebra I",
            "assignments": [
              { "id": "A-0", "name": "Warmup", "points": 10, "dueAt": "2025-10-02T09:00:00-07:00", "checkpointStatus": "Missing", "url": "https://app/assignments/A-0" },
              { "id": "A-1", "name": "Quiz 1", "points": 5, "dueAt": "2025-10-03T09:00:00-07:00", "checkpointStatus": "Submitted", "url": "https://app/assignments/A-1" },
              { "id": "A-2", "name": "Reflection", "points": 25, "dueAt": "2025-10-07T09:00:00-07:00", "checkpointStatus": "Due", "url": "https://app/assignments/A-2" },
              { "id": "A-3", "name": "Problem Set", "points": 20, "dueAt": "2025-10-10T09:00:00-07:00", "checkpointStatus": "Missing", "url": "https://app/assignments/A-3" },
              { "id": "A-4", "name": "Project", "points": 5, "checkpointStatus": "Submitted", "url": "https://app/assignments/A-4" },
              { "id": "A-5", "name": "Next Week Prep", "points": 15, "dueAt": "2025-10-14T09:00:00-07:00", "checkpointStatus": "Due", "url": "https://app/assignments/A-5" }
            ]
          }
        ]
      },
      {
        "id": "S2",
        "name": "Bob",
        "courses": [
          {
            "id": "C-201",
            "name": "Biology",
            "assignments": [
              { "id": "B-1", "name": "Lab 1", "points": 10, "dueAt": "2025-10-10T09:00:00-07:00", "checkpointStatus": "Submitted", "url": "https://app/assignments/B-1" },
              { "id": "B-2", "name": "Worksheet", "points": 15, "checkpointStatus": "Submitted", "url": "https://app/assignments/B-2" },
              { "id": "B-3", "name": "Reading", "points": 10, "dueAt": "2025-10-15T09:00:00-07:00", "checkpointStatus": "Due", "url": "https://app/assignments/B-3" }
            ]
          }
        ]
      }
    ]
  },
  "asOf": "2025-10-08T09:00:00-07:00",
  "timezone": "America/Los_Angeles"
}