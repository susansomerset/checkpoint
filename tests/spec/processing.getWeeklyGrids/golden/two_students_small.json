{
  "S1": {
    "summary": {
      "attentionCounts": {
        "Check": 0,
        "Thumb": 2,
        "Question": 0,
        "Warning": 2
      },
      "totalItems": 4
    },
    "grid": {
      "header": {
        "studentHeader": "Alice ‚Äî ‚ö†Ô∏è:2 / ‚ùì:0 / üëç:2 / ‚úÖ:0",
        "columns": [
          "Class Name",
          "Prior Weeks",
          "Mon (10/6)",
          "Tue (10/7)",
          "Wed (10/8)",
          "Thu (10/9)",
          "Fri (10/10)",
          "Next Week",
          "No Date"
        ]
      },
      "rows": [
        {
          "courseId": "C-101",
          "courseName": "Algebra I",
          "cells": {
            "prior": [
              {
                "id": "A-0",
                "title": "10/2: Warmup (10)",
                "attentionType": "Warning",
                "url": "https://app/assignments/A-0"
              }
            ],
            "weekday": {
              "Mon": [],
              "Tue": [
                {
                  "id": "A-2",
                  "title": "Reflection (25)",
                  "attentionType": "Thumb",
                  "url": "https://app/assignments/A-2"
                }
              ],
              "Wed": [],
              "Thu": [],
              "Fri": [
                {
                  "id": "A-3",
                  "title": "Problem Set (20)",
                  "attentionType": "Warning",
                  "url": "https://app/assignments/A-3"
                }
              ]
            },
            "next": [
              {
                "id": "A-5",
                "title": "Tue: Next Week Prep (15)",
                "attentionType": "Thumb",
                "url": "https://app/assignments/A-5"
              }
            ],
            "noDate": {
              "count": 1,
              "points": 5,
              "label": "1 no due date (5 points)",
              "deepLinkUrl": "https://app/detail?student=S1&course=C-101&nodate=1"
            }
          },
          "summary": {
            "attentionCounts": {
              "Check": 0,
              "Thumb": 2,
              "Question": 0,
              "Warning": 2
            },
            "totalItems": 4
          }
        }
      ]
    }
  },
  "S2": {
    "summary": {
      "attentionCounts": {
        "Check": 1,
        "Thumb": 1,
        "Question": 0,
        "Warning": 0
      },
      "totalItems": 2
    },
    "grid": {
      "header": {
        "studentHeader": "Bob ‚Äî ‚ö†Ô∏è:0 / ‚ùì:0 / üëç:1 / ‚úÖ:1",
        "columns": [
          "Class Name",
          "Prior Weeks",
          "Mon (10/6)",
          "Tue (10/7)",
          "Wed (10/8)",
          "Thu (10/9)",
          "Fri (10/10)",
          "Next Week",
          "No Date"
        ]
      },
      "rows": [
        {
          "courseId": "C-201",
          "courseName": "Biology",
          "cells": {
            "prior": [],
            "weekday": {
              "Mon": [],
              "Tue": [],
              "Wed": [],
              "Thu": [],
              "Fri": [
                {
                  "id": "B-1",
                  "title": "Lab 1 (10)",
                  "attentionType": "Check",
                  "url": "https://app/assignments/B-1"
                }
              ]
            },
            "next": [
              {
                "id": "B-3",
                "title": "Wed: Reading (10)",
                "attentionType": "Thumb",
                "url": "https://app/assignments/B-3"
              }
            ],
            "noDate": {
              "count": 1,
              "points": 15,
              "label": "1 no due date (15 points)",
              "deepLinkUrl": "https://app/detail?student=S2&course=C-201&nodate=1"
            }
          },
          "summary": {
            "attentionCounts": {
              "Check": 1,
              "Thumb": 1,
              "Question": 0,
              "Warning": 0
            },
            "totalItems": 2
          }
        }
      ]
    }
  }
}
